core/chains/starknet/chain.go

```diff
@@ -2,57 +2,121 @@ package starknet
 
 import (
 	"context"
+	"math"
+	"math/rand"
+	"time"
 
+	caigotypes "github.com/dontpanicdao/caigo/types"
+	"github.com/pkg/errors"
+
+	starkChain "github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/chain"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/config"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/db"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/txm"
 	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet"
-	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet/db"
 
 	"github.com/smartcontractkit/sqlx"
 
 	"github.com/smartcontractkit/chainlink/core/chains/starknet/types"
 	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
-var _ starknet.Chain = (*chain)(nil)
+var _ starkChain.Chain = (*chain)(nil)
 
 type chain struct {
 	utils.StartStopOnce
 	id   string
-	cfg  starknet.Config
+	cfg  config.Config
```

Comment:
`config.Config` doesn't sound very informative tbh, previously the config was in the starknet package so that we could reference it and get the idea of where this config is coming from

Reply:
Yeah, we are still looking for the best structure here as discussed.

This is the Chainlink specific top-level config for the StarkNet integration.
The original suggestion was to put it into the `chainlink/starknet` package of the CL StarkNet `relayer` module.